pyqtgraph/graphicsItems/BarGraphItem.py

```diff
@@ -50,119 +44,131 @@ def __init__(self, **opts):
         )
         self._shape = None
         self.picture = None
+        self.bounds = [None, None]  ## caches data bounds
+        self.setData(**opts)
         self.setOpts(**opts)
-        
+
     def setOpts(self, **opts):
         self.opts.update(opts)
         self.picture = None
         self._shape = None
         self.update()
         self.informViewBoundsChanged()
-        
-    def drawPicture(self):
-        self.picture = QtGui.QPicture()
-        self._shape = QtGui.QPainterPath()
-        p = QtGui.QPainter(self.picture)
-        
-        pen = self.opts['pen']
-        pens = self.opts['pens']
-        
-        if pen is None and pens is None:
-            pen = getConfigOption('foreground')
-        
-        brush = self.opts['brush']
-        brushes = self.opts['brushes']
-        if brush is None and brushes is None:
-            brush = (128, 128, 128)
-        
+
+    def setPen(self, *args, **kargs):
+        """Set the pen used to draw the curve."""
+        self.setOpts(pen=fn.mkPen(*args, **kargs))
+
+    def setBrush(self, *args, **kargs):
+        """Set the brush used when filling the area under the curve"""
+        self.setOpts(brush=fn.mkBrush(*args, **kargs))
+
+    def setData(self, **kargs):
```

Comment:
There should be a docstring here with what the expected/supported arguments here are

Reply:
Done